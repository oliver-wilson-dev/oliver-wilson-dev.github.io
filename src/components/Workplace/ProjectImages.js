import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const ProjectImages = ({ projectImages }) => (
  <div className={styles.projectImages}>
    {projectImages.map(({ linkToProject, image: { alt, src } }) => (
      <div key={linkToProject} className={styles.projectImage}>
        <a href={linkToProject}>
          <img alt={alt} src={src} />
        </a>
      </div>
    ))}
  </div>
);

ProjectImages.propTypes = {
  projectImages: PropTypes.arrayOf(
    PropTypes.shape({
      linkToProject: PropTypes.string,
      image: PropTypes.shape({
        alt: PropTypes.string,
        src: PropTypes.string
      })
    })
  )
};

export default ProjectImages;
