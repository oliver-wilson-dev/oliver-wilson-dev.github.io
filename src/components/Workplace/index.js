import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';
import ProjectImages from './ProjectImages';

const Workplace = ({
  companyName,
  content,
  emoji,
  employmentPeriod,
  jobTitle,
  projectImages = []
}) => (
  <React.Fragment>
    <h3 className={styles.companyTitle}>{companyName}</h3>
    <h4 className={styles.jobRole}>
      {jobTitle}
    </h4>
    <h4>{employmentPeriod}</h4>
    {content.map((paragraph, index) => (index === 0
      ? (
        <p key={paragraph}>
          <span className={styles.paragraphEmoji} role="img" aria-label="laptop emoji">{emoji}</span>
          {paragraph}
        </p>
      )
      : <p key={paragraph}>{paragraph}</p>))}
    {!!projectImages.length && <ProjectImages projectImages={projectImages} />}
  </React.Fragment>
);

export default Workplace;

Workplace.propTypes = {
  companyName: PropTypes.string.isRequired,
  content: PropTypes.array.isRequired,
  emoji: PropTypes.string.isRequired,
  employmentPeriod: PropTypes.string.isRequired,
  jobTitle: PropTypes.string.isRequired,
  projectImages: PropTypes.array
};
