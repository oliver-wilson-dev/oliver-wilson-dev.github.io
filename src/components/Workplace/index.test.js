import React from 'react';
import { shallow } from 'enzyme';
import Workplace from '.';

jest.mock('./ProjectImages', () => {
  const ProjectImages = () => null;

  return ProjectImages;
});

const defaultProps = {
  companyName: 'test-company-name',
  content: [],
  emoji: 'test-emoji',
  employmentPeriod: 'test-employment-period',
  jobTitle: 'test-job-title',
  projectImages: []
};


const render = ({ overrideProps = {} } = { overrideProps: {} }) => shallow(
  <Workplace
    {...defaultProps}
    {...overrideProps}
  />
);

describe('Workplace', () => {
  describe('when there is no content', () => {
    describe('when there are no projectImages', () => {
      it('should render correctly', () => {
        expect(render()).toMatchSnapshot();
      });
    });

    describe('when there are projectImages', () => {
      it('should render correctly', () => {
        expect(render({
          overrideProps: {
            projectImages: ['test-project-image']
          }
        })).toMatchSnapshot();
      });
    });
  });

  describe('when there is content', () => {
    describe('when there are no projectImages', () => {
      it('should render correctly', () => {
        expect(render({
          overrideProps: {
            content: ['test-content']
          }
        })).toMatchSnapshot();
      });
    });

    describe('when there are projectImages', () => {
      it('should render correctly', () => {
        expect(render({
          overrideProps: {
            projectImages: ['test-project-image'],
            content: ['test-content']
          }
        })).toMatchSnapshot();
      });
    });
  });
});
