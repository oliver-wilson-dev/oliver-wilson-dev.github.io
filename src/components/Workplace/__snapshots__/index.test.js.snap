// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workplace when there is content when there are no projectImages should render correctly 1`] = `
<Fragment>
  <h3
    className="companyTitle"
  >
    test-company-name
  </h3>
  <h4
    className="jobRole"
  >
    test-job-title
  </h4>
  <h4>
    test-employment-period
  </h4>
  <p
    key="test-content"
  >
    <span
      aria-label="an test-emoji emoji"
      className="paragraphEmoji"
      role="img"
    >
      test-emoji
    </span>
    test-content
  </p>
</Fragment>
`;

exports[`Workplace when there is content when there are projectImages should render correctly 1`] = `
<Fragment>
  <h3
    className="companyTitle"
  >
    test-company-name
  </h3>
  <h4
    className="jobRole"
  >
    test-job-title
  </h4>
  <h4>
    test-employment-period
  </h4>
  <p
    key="test-content"
  >
    <span
      aria-label="an test-emoji emoji"
      className="paragraphEmoji"
      role="img"
    >
      test-emoji
    </span>
    test-content
  </p>
  <ProjectImages
    projectImages={
      Array [
        "test-project-image",
      ]
    }
  />
</Fragment>
`;

exports[`Workplace when there is no content when there are no projectImages should render correctly 1`] = `
<Fragment>
  <h3
    className="companyTitle"
  >
    test-company-name
  </h3>
  <h4
    className="jobRole"
  >
    test-job-title
  </h4>
  <h4>
    test-employment-period
  </h4>
</Fragment>
`;

exports[`Workplace when there is no content when there are projectImages should render correctly 1`] = `
<Fragment>
  <h3
    className="companyTitle"
  >
    test-company-name
  </h3>
  <h4
    className="jobRole"
  >
    test-job-title
  </h4>
  <h4>
    test-employment-period
  </h4>
  <ProjectImages
    projectImages={
      Array [
        "test-project-image",
      ]
    }
  />
</Fragment>
`;
